docs/developer/architecture/logs.rst

```diff
@@ -197,6 +197,422 @@ overview page.
 A Grafana dashboard gathering health/performance KPIs, as well as alerts for
 log centralization service is available when deploying/upgrading MetalK8s.
 
+Components
+----------
+
+Our Log Centralization system can be splitted into several components as
+follows:
+
+.. parsed-literal::
+
+  +-----------+     +-------------+     +----------+     +---------+
+  | Collector | ==> | Distributor | ==> | Ingester | <=> | Querier |
+  +-----------+     +-------------+     +----------+     +---------+
+
+.. todo:: do a better schema. :)
+
+Collector
+~~~~~~~~~
+
+The collector is responsible for processing logs from all the sources
+(files, journal, containers, ...), enriching the logs with metadata (labels)
+coming from parsing/filtering them or by querying external sources such as
+APIs, then finally forwarding these logs to one or multiple distributors.
+
+Distributor (Router)
+~~~~~~~~~~~~~~~~~~~~
+
+The distributor is the component that receives incoming streams from the
+collectors, validates and prepares the streams for storage, then forwards
+them to the right ingester.
+The distributor can also do some processing on the streams to enrich them
+with metadata (labels).
+
+Ingester (Storage)
+~~~~~~~~~~~~~~~~~~
+
+The job of the ingester is to write log streams on persistent storage such
+as a filesystem or a database, but also to return the data requested by
```

Comment:
in Loki ecosystem Ingesters only return data that have not been flashed to storage they re not used to read the data back,no?

Reply:
true, it's not clear here.